aea/cli/registry/publish.py

```diff
@@ -76,13 +76,14 @@ def publish_agent(ctx: Context):
         "skills": ctx.agent_config.skills,
     }
 
-    files = {"file": open(output_tar, "rb")}
-    if is_readme_present(readme_source_path):
-        files["readme"] = open(readme_source_path, "rb")
+    with open(output_tar, "rb") as fd:
+        files = {"file": fd}
+        if is_readme_present(readme_source_path):
+            files["readme"] = open(readme_source_path, "rb")
```

Comment:
better to do nested with? i.e. another context manager for readme_source_path?

Reply:
missed this, thanks, but this readme  is not removed